### ISSUE TITLE
Plotting spectrum fails when y_scale lower than -20

### ISSUE THREAD

**Issue by ThomasHerard** [ACCarnall/bagpipes#34]
I encountered this issue when plotting spectra with small fluxes in the order of 10**(-20). Once `y_scale <= -20`, the arrays' `dtype` becomes `object` and the following error is raised: 

```
TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

Quick fix would be to cast `y_scale` to float before plotting, I will open a pull request where I found the issue but it might be found elsewhere as well.

**Comment by ACCarnall** [comment]
Hi Thomas,

I'm not convinced this is related to any specific values of y_scale. I've had this kind of error before when the spectrum that's been provided to the code contains significantly higher flux values than the associated photometry. Might this be what's going on in your case? As I often tell people, I see the plotting functions as kind of an add-on to the code rather than core functionality. If they work, great, but I can't guarantee they will in all situations. I'd always encourage people to make their own plots to check their outputs.

Cheers,
Adam

**Comment by ThomasHerard** [comment]
It does depend on the `y_scale` value but seems system dependant as explained [here](https://stackoverflow.com/questions/74573624/why-does-the-dtype-of-a-numpy-array-automatically-change-to-object-if-you-mult) on stackoverflow, I have had the switch to object type at 10^20 or 10^(-20) on two machines, which for latter can be inconvenient as those fluxes scales are common.

However, I understand your policy on the plotting functions and have not opened a pull request, thanks for the reply anyway!

Cheers,
Thomas

**Comment by ACCarnall** [comment]
Ah ok that's interesting, thanks. Feel free to open a pull request to fix this if you'd like, no pressure though.

**Comment by ACCarnall** [comment]
I believe this should now be fixed in v1.0.2
